Plugins/GitUIPluginInterfaces/IGitModule.cs

```diff
@@ -70,6 +74,21 @@ public interface IGitModule
         /// <summary>Indicates HEAD is not pointing to a branch (i.e. it is detached).</summary>
         bool IsDetachedHead();
 
+        /// <summary>
+        /// Convert the path for the Git executable.
+        /// For WSL Git, the path will be adjusted.
+        /// </summary>
+        /// <param name="path">Path as seen by the Git Extensions native (Windows) executable.</param>
+        /// <returns>Posix path if Windows Git, WSL path for WSL Git.</returns>
+        public string GetPathForGitExec(string? path);
+
+        /// <summary>
+        /// Convert a path to native (Windows) format.
+        /// </summary>
+        /// <param name="path">Path as seen by the Git executable, possibly WSL Git.</param>
+        /// <returns>Path in in Windows format with native file separators.</returns>
```

Comment:
Does it return an empty string for an invalid path or throws?

Reply:
Empty if null path, set path separators to Windows if wslDistro is empty/null, otherwise try to convert the known path formats in WSL to Windows